src/draw/drawCircleOnImage.ts

```diff
@@ -60,20 +61,22 @@ export function drawCircleOnImage(
   radius = Math.round(radius);
 
   if (radius === 0) {
-    newImage.setVisiblePixel(center.column, center.row, color);
+    setBlendedVisiblePixel(newImage, center.column, center.row, { color });
     return newImage;
   }
 
   if (!fill) {
     circle(center.column, center.row, radius, (column: number, row: number) => {
-      newImage.setVisiblePixel(column, row, color);
+      setBlendedVisiblePixel(newImage, column, row, { color });
     });
   } else {
     if (radius === 1) {
-      newImage.setVisiblePixel(center.column, center.row, fill);
+      setBlendedVisiblePixel(newImage, center.column, center.row, {
+        color: fill,
+      });
     }
     circle(center.column, center.row, radius, (column: number, row: number) => {
-      newImage.setVisiblePixel(column, row, color);
+      setBlendedVisiblePixel(newImage, column, row, { color });
```

Comment:
You can create the color object outside of the callback to avoid recreating it for every pixel.